pkg/notes/notes_v2.go

```diff
@@ -0,0 +1,249 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notes
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+
+	"k8s.io/release/pkg/notes/options"
+
+	"github.com/cheggaaa/pb/v3"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	gitobject "github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/nozzle/throttler"
+	"github.com/sirupsen/logrus"
+)
+
+type commitPrPair struct {
+	Commit *gitobject.Commit
+	PrNum  int
+}
+
+type releaseNotesAggregator struct {
+	releaseNotes *ReleaseNotes
+	sync.RWMutex
+}
+
+func (g *Gatherer) ListReleaseNotesV2() (*ReleaseNotes, error) {
+	// left parent of Git commits is always the main branch parent
+	pairs, err := g.listLeftParentCommits(g.options)
+	if err != nil {
+		return nil, errors.Wrap(err, "listing offline commits")
+	}
+
+	// load map providers specified in options
+	mapProviders := []MapProvider{}
+	for _, initString := range g.options.MapProviderStrings {
+		provider, err := NewProviderFromInitString(initString)
+		if err != nil {
+			return nil, errors.Wrap(err, "while getting release notes map providers")
+		}
+		mapProviders = append(mapProviders, provider)
+	}
+
+	t := throttler.New(maxParallelRequests, len(pairs))
+
+	aggregator := releaseNotesAggregator{
+		releaseNotes: NewReleaseNotes(),
+	}
+
+	pairsCount := len(pairs)
+	logrus.Infof("processing release notes for %d commits", pairsCount)
+	bar := pb.Full.Start(pairsCount)
```

Comment:
hmm this is what have been stuck in my mind too -- how is it running in release process log? I imagine this can be disabled through `isatty` or something, but at the same time I'm curious if it'd be simpler to just make it different (`stdout` vs `stderr`) from the logrus output.

Reply:
When a new patch release is cut, krel generates the changelog automatically:
https://github.com/kubernetes/release/blob/master/pkg/anago/stage.go#L465-L484
(It's not the same as the md draft which is copied from k/sig-release)
This is run in a Cloud Build job and all output (stderr and stdout) is sent to the log which is archived along with the release artifacts.

Reply:
I've made it only show up if it's TTY and it will go to stdout, just like the prompts under `--fix`.